### ISSUE TITLE
[BUG] Update validation check in `qml.Hermitian` to support multiple interfaces

### ISSUE THREAD

**Issue by antalszava** [PennyLaneAI/pennylane#2132]
### Expected behavior

The following executes correctly:
```python
import pennylane as qml
import numpy as np
import tensorflow as tf

dev = qml.device('default.qubit', wires=1, shots=10)

arr = np.array(
    [
        1.32,
        2.312,
    ]
)

herm = tf.Variable(np.outer(arr, arr))

@qml.qnode(dev, interface='tf')
def circuit():
    return qml.sample(qml.Hermitian(herm, wires=0))

res = circuit()
```

### Actual behavior

`AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'conj'` is raised due to the

```python
        if not qml.math.allclose(A, A.conj().T):
            raise ValueError("Observable must be Hermitian.")
```
validation check.

### Additional information

To resolve the issue, it may be good to consider if we could support `conj` and/or `T` in `qml.math` for multiple interfaces.

On top of that, however, `qml.math.allclose` will break jittability, hence potentially omitting the validation as a whole should also be considered.

### Tracebacks

```shell
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-0a0e7a84a095> in <module>
     21 print(res, type(res), res.dtype)
     22 
---> 23 circuit.qtape.get_output_domain()

~/xanadu/pennylane/pennylane/tape/tape.py in get_output_domain(self)
   1079 
   1080                 output_domain = (
-> 1081                     int if all(np.issubdtype(e.dtype, int) for e in observable.eigvals) else float
   1082                 )
   1083 

~/xanadu/pennylane/pennylane/measure.py in eigvals(self)
    144         if self.obs is not None:
    145             try:
--> 146                 return self.obs.eigvals
    147             except NotImplementedError:
    148                 pass

~/xanadu/pennylane/pennylane/ops/qubit/observables.py in eigvals(self)
    102             array: array containing the eigenvalues of the Hermitian observable
    103         """
--> 104         return self.eigendecomposition["eigval"]
    105 
    106     def diagonalizing_gates(self):

~/xanadu/pennylane/pennylane/ops/qubit/observables.py in eigendecomposition(self)
     83             dict[str, array]: dictionary containing the eigenvalues and the eigenvectors of the Hermitian observable
     84         """
---> 85         Hmat = self.matrix
     86         Hmat = qml.math.to_numpy(Hmat)
     87         Hkey = tuple(Hmat.flatten().tolist())

~/xanadu/pennylane/pennylane/operation.py in matrix(self)
    342             array: matrix representation
    343         """
--> 344         return self._matrix(*self.parameters)
    345 
    346     @classmethod

~/xanadu/pennylane/pennylane/ops/qubit/observables.py in _matrix(cls, *params)
     66             raise ValueError("Observable must be a square matrix.")
     67 
---> 68         if not qml.math.allclose(A, A.conj().T):
     69             raise ValueError("Observable must be Hermitian.")
     70 

~/anaconda3/lib/python3.8/site-packages/tensorflow/python/framework/ops.py in __getattr__(self, name)
    440         from tensorflow.python.ops.numpy_ops import np_config
    441         np_config.enable_numpy_behavior()""".format(type(self).__name__, name))
--> 442     self.__getattribute__(name)
    443 
    444   @staticmethod

AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'conj'
```

### System information

```shell
Name: PennyLane
Version: 0.21.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/antal/xanadu/pennylane
Requires: numpy, scipy, networkx, retworkx, autograd, toml, appdirs, semantic_version, autoray, cachetools, pennylane-lightning
Required-by: PennyLane-Cirq, amazon-braket-pennylane-plugin, PennyLane-Orquestra, pennylane-qulacs, PennyLane-Honeywell, PennyLane-qiskit, PennyLane-AQT, PennyLane-PQ, PennyLane-Forest, PennyLane-qsharp, PennyLane-Qchem, PennyLane-IonQ, PennyLane-SF, PennyLane-Lightning
Platform info:           Linux-5.13.0-27-generic-x86_64-with-glibc2.10
Python version:          3.8.5
Numpy version:           1.19.2
Scipy version:           1.7.3
Installed devices:
- cirq.mixedsimulator (PennyLane-Cirq-0.19.0)
- cirq.pasqal (PennyLane-Cirq-0.19.0)
- cirq.qsim (PennyLane-Cirq-0.19.0)
- cirq.qsimh (PennyLane-Cirq-0.19.0)
- cirq.simulator (PennyLane-Cirq-0.19.0)
- braket.aws.qubit (amazon-braket-pennylane-plugin-1.5.3)
- braket.local.qubit (amazon-braket-pennylane-plugin-1.5.3)
- orquestra.forest (PennyLane-Orquestra-0.15.0)
- orquestra.ibmq (PennyLane-Orquestra-0.15.0)
- orquestra.qiskit (PennyLane-Orquestra-0.15.0)
- orquestra.qulacs (PennyLane-Orquestra-0.15.0)
- qulacs.simulator (pennylane-qulacs-0.17.0.dev0)
- honeywell.hqs (PennyLane-Honeywell-0.16.0.dev0)
- qiskit.aer (PennyLane-qiskit-0.18.0.dev0)
- qiskit.basicaer (PennyLane-qiskit-0.18.0.dev0)
- qiskit.ibmq (PennyLane-qiskit-0.18.0.dev0)
- aqt.noisy_sim (PennyLane-AQT-0.18.0)
- aqt.sim (PennyLane-AQT-0.18.0)
- projectq.classical (PennyLane-PQ-0.18.0.dev0)
- projectq.ibm (PennyLane-PQ-0.18.0.dev0)
- projectq.simulator (PennyLane-PQ-0.18.0.dev0)
- forest.numpy_wavefunction (PennyLane-Forest-0.18.0.dev0)
- forest.qvm (PennyLane-Forest-0.18.0.dev0)
- forest.wavefunction (PennyLane-Forest-0.18.0.dev0)
- microsoft.QuantumSimulator (PennyLane-qsharp-0.19.0)
- ionq.qpu (PennyLane-IonQ-0.17.0.dev0)
- ionq.simulator (PennyLane-IonQ-0.17.0.dev0)
- strawberryfields.fock (PennyLane-SF-0.20.0.dev0)
- strawberryfields.gaussian (PennyLane-SF-0.20.0.dev0)
- strawberryfields.gbs (PennyLane-SF-0.20.0.dev0)
- strawberryfields.remote (PennyLane-SF-0.20.0.dev0)
- strawberryfields.tf (PennyLane-SF-0.20.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.21.0.dev0)
- default.gaussian (PennyLane-0.21.0.dev0)
- default.mixed (PennyLane-0.21.0.dev0)
- default.qubit (PennyLane-0.21.0.dev0)
- default.qubit.autograd (PennyLane-0.21.0.dev0)
- default.qubit.jax (PennyLane-0.21.0.dev0)
- default.qubit.tf (PennyLane-0.21.0.dev0)
- default.qubit.torch (PennyLane-0.21.0.dev0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by timmysilv** [comment]
duplicate of #2892, resolved
